src/nvim/hardcopy.c

```diff
@@ -2117,7 +2117,7 @@ int mch_print_init(prt_settings_T *psettings, char_u *jobname, int forceit)
   props = enc_canon_props(p_encoding);
   if (!(props & ENC_8BIT) && ((*p_pmcs != NUL) || !(props & ENC_UNICODE))) {
     p_mbenc_first = NULL;
-    int effective_cmap;
+    int effective_cmap = 0;
```

Comment:
Argg, I can't comment on unchanged line.
Well, on line 2133, changing order of conditions from `if (p_mbchar == NULL && p_mbenc_first != NULL) {` to the opposite, clears the warning without the variable initialization.